astropy/table/tests/test_table.py

```diff
@@ -1364,3 +1372,92 @@ def test_nested_iteration():
         for r2 in t:
             out.append((r1['a'], r2['a']))
     assert out == [(0, 0), (0, 1), (1, 0), (1, 1)]
+
+
+class TestPandas(object):
+
+    @pytest.mark.skipif('not HAS_PANDAS')
+    def test_simple(self):
+
+        from pandas import DataFrame
+
+        t = table.Table()
+
+        for endian in ['<', '>']:
+            for kind in ['f', 'i']:
+                for byte in ['2','4','8']:
+                    dtype = np.dtype(endian + kind + byte)
+                    x = np.array([1,2,3], dtype=dtype)
+                    t[endian + kind + byte] = x
+
+        t['u'] = ['a','b','c']
+        t['s'] = [b'a', b'b', b'c']
+
+        d = t.to_pandas()
+
+        for column in t.columns:
+            if column == 'u':
+                assert np.all(t['u'] == np.array(['a','b','c']))
+            elif column == 's':
+                assert np.all(t['s'] == np.array([b'a',b'b',b'c']))
+            else:
+                assert_allclose(t[column], d[column])
```

Comment:
Passing a bytes array does work:

```
In [4]: d['a'] = np.array(['a','b','c'],dtype='S1')

In [5]: d['a'].dtype
Out[5]: dtype('S1')
```

but not if passed as part of a dictionary:

```
In [9]: d2 = DataFrame({'a':np.array(['a','b','c'],dtype='S1'),'b':np.array(['a','b','c'],dtype='U1')})

In [10]: d2
Out[10]: 
      a  b
0  b'a'  a
1  b'b'  b
2  b'c'  c

In [11]: d2['a'].dtype
Out[11]: dtype('O')
```

:panda_face: :question: :question: